_posts/2019-06-21-install-esxi-using-foreman-using-ipxe-bootstrapping.md

```diff
@@ -0,0 +1,209 @@
+---
+layout: post
+title: 'Install ESXi using Foreman using iPXE bootstrapping'
+date: 2019-06-21 14:25:45
+author:
+  name: Amardeep Kahali
+tags:
+- foreman
+- esxi
+- vmware
+- ipxe
+- bootstrapping
+- uefi
+guest_post: true
+---
+
+This post is intended to simplify the ESXi installation process [described in the previous post](https://theforeman.org/2018/08/deploying-esxi-through-foreman.html) using the new iPXE bootstrapping feature of Foreman introduced in v1.20.
+
+### Requirements
+
+* A fully working Foreman instance. Follow the [manual](https://theforeman.org/manuals/latest/index.html#3.InstallingForeman) to install Foreman. After installation, please configure Foreman so that a Linux OS (for e.g. Centos) can be deployed successfully through Foreman.
+
+* An [ESXi ISO](https://my.vmware.com/en/group/vmware/evalcenter). For this demo we will use `VMware-VMvisor-Installer-6.7.0-8169922.x86_64.iso` but the same process will work on any version of ESXi.  
+
+Note: This demo is run on a Ubuntu system. 
+
+### 1: Enable iPXE in Foreman
+
+---
+
+Since `foreman-installer` doesn't support creating an iPXE based Foreman instance yet, we need to make some changes to the DHCP configuration manually to enable iPXE. The `dhcpd.conf` should look like this:
+
+    ```
+    # Bootfile Handoff
+    next-server ${foreman_server_fqdn};
+    option architecture code 93 = unsigned integer 16 ;
+
+    if exists user-class and option user-class = "iPXE" {
+      filename "http://${foreman_server_fqdn}/unattended/iPXE?bootstrap=1";
+    } elsif option architecture = 00:06 {
+      filename "ipxe.efi";
+    } elsif option architecture = 00:07 {
+      filename "snp.efi";
+    } elsif option architecture = 00:09 {
+      filename "ipxe.efi";
+    } else {
+      filename "undionly.kpxe";
+    }
+
+    ```
+
+**Note:** For both Legacy BIOS and UEFI configs, we are using the iPXE bootloaders that use the iPXE drivers of the network card (undionly.kpxe/snp.efi) as we faced issues with the native drivers (ipxe.pxe/ipxe.efi) on our hardware (Dell PowerEdge 14G with Broadcom NICs).  
+References: [1](http://forum.ipxe.org/showthread.php?tid=5806), [2](https://github.com/warewulf/warewulf3/issues/84)
+
+
+Now let's build the iPXE bootloaders. In this example, the ISO images will be hosted on the Foreman server itself. Since HTTP is already used, we have to use FTP to expose the images. So the iPXE bootloaders need to have FTP support. If you use a separate HTTP based image repository, you may skip that step.
```

Comment:
This paragraph is little bit hard to understand, can you explain why exactly do we need FTP?

Reply:
If we want to host the ISO images on the Foreman server itself, what protocol would we use?
HTTP is used by foreman already...although there is a hacky way to serve another vhost on FOreman itself, it's better to just use FTP